src/controllers/product.controller.js

```diff
@@ -0,0 +1,29 @@
+import { ProductService } from '../services/product.service';
+
+export const createProduct = async (req, res) => {
+  const images = req.images;
+  try {
+    const { name, price, categoryId, expiryDate, bonus } = req.body;
+    const userId = req.user.id;
+    const productInfo = {
+      name,
+      price,
+      categoryId,
+      expiryDate,
+      bonus,
+      images,
+      userId,
+    };
```

Comment:
Since we have validation, you can refactor this code by using the spread operator. It's not necessary to mention the fields.